tests/ci/lambda/src/bin/purge-stale-builds.rs

```diff
@@ -115,6 +116,22 @@ async fn handle(_event: LambdaEvent<Value>) -> Result<(), Error> {
     let mut ec2_terminated_instances: Vec<String> = vec![];
     let mut stopped_builds: u64 = 0;
 
+    let now_as_secs = DateTime::from(SystemTime::now()).secs();
+    // Instances do not properly shut down from time to time. Gather a list of all hanging ec2
+    // instances longer than 2 hours that fall under that umbrella.
+    if let Some(ref ec2_describe_response) = ec2_describe_response_optional {
```

Comment:
That's already done above when we ask for the result on L97-108, there are logs that emit if the command has failed.

Reply:
I don't see any logging around this:
```
        } else {
            None
        };
```

The logging is around `ec2_client.describe_instances()` returning an `Err`.

Reply:
That's not necessarily an error. That simply handles the case where ec2_client_optional is `None`.
`ec2_client_optional` is allowed to be `None` since all the other CI dimensions share the same lambda code. The other CI dimensions do not have ec2 permissions, so their lambdas will have `None` for `ec2_client_optional`.
This commit https://github.com/aws/aws-lc/commit/0fb89012102f62d2db4cb022fb9e44f0287e7d82 has more details regarding this.